### PR TITLE
UX: Add "filter for more" to icon picker

### PR DESCRIPTION
**Pull request by pmusaraj**
Our icon picker maxes out at 200 results at a time, this PR uses the existing "filter for more" row to clarify that the listed results are incomplete.

image

### CODE REVIEW

File: app/assets/javascripts/select-kit/addon/components/filter-for-more.hbs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{{#if this.collection.content.shouldShowMoreTip}}
+  <div class="filter-for-more">
+    {{i18n "select_kit.components.filter_for_more"}}
+  </div>
+{{/if}}
```
Comment (CvX) [inline app/assets/javascripts/select-kit/addon/components/filter-for-more.hbs:5]: Can we make this into a glimmer/gjs component?

```suggestion
import i18n from "discourse-common/helpers/i18n";

const FilterForMore = <template>
  {{#if @collection.content.shouldShowMoreTip}}
    <div class="filter-for-more">
      {{i18n "select_kit.components.filter_for_more"}}
    </div>
  {{/if}}
</template>

export default FilterForMore;
```

File: app/assets/javascripts/select-kit/addon/components/icon-picker.js [hunk 2]
```diff
@@ -18,22 +23,36 @@ export default MultiSelectComponent.extend({
...
     if (isDevelopment()) {
       disableMissingIconWarning();
     }
+
+    this.insertAfterCollection(MAIN_COLLECTION, MORE_ICONS_COLLECTION);
+  },
+
+  modifyComponentForCollection(collection) {
+    if (collection === MORE_ICONS_COLLECTION) {
+      return "filter-for-more";
```
Comment (CvX) [inline app/assets/javascripts/select-kit/addon/components/icon-picker.js:37]: Is it possible to import FilterForMore and use the reference here? Or does it break somewhere on the say to the `{{component}}` helper? 😃

File: app/assets/javascripts/select-kit/addon/components/filter-for-more.hbs (hunk 1 above)
Comment (pmusaraj) [reply app/assets/javascripts/select-kit/addon/components/filter-for-more.hbs]: Yes, done!

File: app/assets/javascripts/select-kit/addon/components/icon-picker.js (hunk 2 above)
Comment (pmusaraj) [reply app/assets/javascripts/select-kit/addon/components/icon-picker.js]: Yes, this works too.

**Comment by pmusaraj** [comment]
This should be ready to go now @CvX, if you can have another look. Thanks!

**Review by CvX — APPROVED** [review]

**Pull request merged**